### PR TITLE
refactor: Refine Notes and Checklist Component to Match Updated Design

### PR DESCRIPTION
**Pull request by hassangulzar1**
https://www.loom.com/share/11c4e06bf2d9406c9bbc904c5d8e746b?sid=9a65c2af-3120-4ce9-a249-c528d8b47a9c

### CODE REVIEW

File: components/AudioRecorder/index.tsx [hunk 1]
```diff
@@ -189,21 +286,39 @@ export const AudioRecorderComponent: React.FC = () => {
     reset();
   };
 
+  console.log(sidebarContent);
```
Comment (dogg3) [inline components/AudioRecorder/index.tsx:289]: remove this!

File: components/AudioRecorder/index.tsx [hunk 2]
```diff
@@ -244,7 +359,7 @@ export const AudioRecorderComponent: React.FC = () => {
         <WhiteSpace height={'32px'} />
         <div className="flex flex-col items-center">
           {!recorderControls.isRecording ? (
-            // IS NOT RECORDING
+            //! IS NOT RECORDING
```
Comment (dogg3) [inline components/AudioRecorder/index.tsx:362]: Why having ! infront. Remov!

File: components/AudioRecorder/index.tsx [hunk 3]
```diff
@@ -285,9 +400,11 @@ export const AudioRecorderComponent: React.FC = () => {
               </div>
             </div>
           ) : (
-            // is Recording
+            //! is Recording
```
Comment (dogg3) [inline components/AudioRecorder/index.tsx:403]: Same here1

File: components/AudioRecorder/index.tsx [hunk 4]
```diff
@@ -59,7 +88,72 @@ export const AudioRecorderComponent: React.FC = () => {
...
+    }
+  };
+
+  // !Function to close sidebar
+  const closeSidebar = () => {
+    setIsSidebarExpanded(false);
+    setSidebarContent(null);
+  };
+
+  //! Side states for responsive Design
```
Comment (dogg3) [inline components/AudioRecorder/index.tsx:116]: This comment is a little bit weird!

File: components/AudioRecorder/index.tsx [hunk 5]
```diff
@@ -59,7 +88,72 @@ export const AudioRecorderComponent: React.FC = () => {
...
+  };
+
+  const handleModalCancel = () => {
+    setIsModalVisible(false);
+    setSidebarContent(null);
+  };
 
+  // !Effect to handle clicking outside the sidebar
+
+  useEffect(() => {
```
Comment (dogg3) [inline components/AudioRecorder/index.tsx:133]: `useEffect(() => {
  const handleClickOutside = (event: MouseEvent) => {
    const clickedOutsideSidebar = sidebarRef.current && !sidebarRef.current.contains(event.target as Node);
    const clickedOutsideModal = modalRef.current && !modalRef.current.contains(event.target as Node);
    
    if (isSidebarExpanded && clickedOutsideSidebar && (!isModalVisible || clickedOutsideModal)) {
      closeSidebar();
    }
  };

  document.addEventListener("mousedown", handleClickOutside);
  return () => document.removeEventListener("mousedown", handleClickOutside);
}, [isSidebarExpanded, isModalVisible, sidebarRef, modalRef]);`

Comment (dogg3) [inline components/AudioRecorder/index.tsx:133]: Try to make this code cleaner for readabillity:


```suggestion

useEffect(() => {
  const handleClickOutside = (event: MouseEvent) => {
    if (!isSidebarExpanded) return;

    const clickedOutsideSidebar =
      sidebarRef.current && !sidebarRef.current.contains(event.target as Node);

    const clickedOutsideModal =
      isModalVisible &&
      modalRef.current &&
      !modalRef.current.contains(event.target as Node);

    if (clickedOutsideSidebar && !clickedOutsideModal) {
      closeSidebar();
    }
  };

  // Add event listener for mouse clicks
  document.addEventListener('mousedown', handleClickOutside);

  // Cleanup event listener
  return () => {
    document.removeEventListener('mousedown', handleClickOutside);
  };
}, [isSidebarExpanded, isModalVisible, sidebarRef, modalRef, closeSidebar]);
```

**Review by dogg3 — CHANGES_REQUESTED** [review]
Overall awesome job @hassangulzar1.

The code is overall well-structured and functional, but there are opportunities for improvement that could enhance readability, reduce complexity, and make it cleaner.

Refactor Sidebar: Consider refactoring the sidebar logic into a separate component. This will help modularize the code, making it more readable and maintainable, especially as the sidebar functionality grows.

Handle Click Outside Sidebar: The handleClickOutside function can be cleaned up by returning early to avoid deep nesting. Additionally, using memoization for stable dependencies (such as sidebarRef, modalRef, and closeSidebar) will ensure that the function is not unnecessarily recreated, improving performance.

These changes would make the code more modular and easier to manage, while also enhancing performance and reducing cognitive load when reading the code.

File: components/AudioRecorder/index.tsx [hunk 6]
```diff
@@ -342,39 +459,145 @@ export const AudioRecorderComponent: React.FC = () => {
         </div>
         <WhiteSpace height={'38px'} />
       </div>
+
+      {/* Full-height Sidebar */}
       <div
```
[the remark this answers is not in the document]
Comment (hassangulzar1) [reply components/AudioRecorder/index.tsx:464]: ok ok wait i will correct it

File: components/AudioRecorder/index.tsx (hunk 4 above)
Comment (hassangulzar1) [reply components/AudioRecorder/index.tsx:116]: ok ok i will keep it cleaner

**Pull request closed without merging**